### PR TITLE
DOC: Switch to https where used

### PR DESCRIPTION
**Pull request by bashtage**
Switch links to https

Quite big but 100% URL clean up (and one tiny format in conf.py for the intersphinc dict).  ALmost all http://->https://.  Standardized on raw.githubusercontent.com, the preferred URI, for Rdatasets .

### CODE REVIEW

File: statsmodels/tsa/statespace/tests/results/test_sarimax_stata.do [hunk 1]
```diff
@@ -1,5 +1,5 @@
 // Example 1: ARIMA model
-use http://www.stata-press.com/data/r12/wpi1, clear
+use https://www.stata-press.com/data/r12/wpi1, clear
```
Comment (josef-pkt) [inline statsmodels/tsa/statespace/tests/results/test_sarimax_stata.do]: I need to check if Stata actually uses https or whether it redirects as needed or needs http
here they use http links
https://www.stata-press.com/data/r12/mv.html

Comment (josef-pkt) [reply statsmodels/tsa/statespace/tests/results/test_sarimax_stata.do]: Stata 14 which I have on this computer, still uses `use http://...` in the documentation (docstring for `use`).
However, both http and https addresses work to download the data inside Stata interpreter.

Comment (bashtage) [reply statsmodels/tsa/statespace/tests/results/test_sarimax_stata.do]: Stata supports https

https://www.stata-press.com/data/r12/wpi1.dta

Comment (bashtage) [reply statsmodels/tsa/statespace/tests/results/test_sarimax_stata.do]: No redrect

image

**Comment by josef-pkt** [comment]
LGTM

**Comment by jbrockmendel** [comment]
+1

**Pull request merged**